### PR TITLE
Add support for Gitea in SCM/CI integration

### PR DESCRIPTION
**Pull request by krauselukas**
*TODO:*

- [x] Add specs
   - [x] scm_extractor_spec.rb
   - [x] gitea_status_reporter.rb (@eduardoj)
   - [x] scm_webhook_spec.rb
   - [x] trigger_workflow_controller_spec.rb
   - [x] yaml_downloader_spec.rb
   - [x] api/gitea_api/v1/client_spec.rb (?) 
- [x] Handle failed HTTP requests in API wrapper for Gitea
- [x] Metrics, Exception Handling and Updating WorkflowRun in GiteaStatusReporter
- [x] Final manual testing through a full setup with Gitea -> partially done already

There is a lot of repeated code, but I want to first have the integration working and tackle the refactoring afterwards. There were a lot of TODO'S for refactorings already. It will be easier to do it after everything is working

### CODE REVIEW

**Comment by krauselukas** [comment]
[Support Gitea (L)](https://trello.com/c/S4WdyCJW/1974-support-gitea-l)

File: src/api/app/apis/gitea_api/v1/client.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      def initialize(api_endpoint:, token:)
+        @api_endpoint = api_endpoint + 'api/v1/'
+        @token = token
+      end
+
+      # owner: owner of the repository
+      # repo: name of the repository
+      # sha: sha of the commit
+      # https://try.gitea.io/api/swagger#/repository/repoCreateStatus
+      def create_commit_status(owner:, repo:, sha:, state:, **kwargs)
```
Comment (danidoni) [inline src/api/app/apis/gitea_api/v1/client.rb:27]:
```suggestion
      def create_commit_status(owner:, repo_name:, sha:, state:, **kwargs)
```

Comment (krauselukas) [reply src/api/app/apis/gitea_api/v1/client.rb:27]: I wanted to stick with the naming used in the api docs of gitea. does this makes sense?

File: src/api/app/apis/gitea_api/v1/client.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+module GiteaAPI
```
Comment (hennevogel) [inline src/api/app/apis/gitea_api/v1/client.rb:1]: What is `app/apis`?

File: src/api/app/services/trigger_controller_service/scm_extractor.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
-# TODO: Extract SCM-specific code to separate classes. This class is too big for GitHub and GitLab.
+# TODO: Extract SCM-specific code to separate classes. This class is too big for GitHub, GitLab and Gitea.
+# rubocop:disable Metrics/ClassLength
```
Comment (hennevogel) [inline src/api/app/services/trigger_controller_service/scm_extractor.rb:2]: We have the rubocop TODO file for this.

File: src/api/app/apis/gitea_api/v1/client.rb (hunk 1 above)
Comment (danidoni) [reply src/api/app/apis/gitea_api/v1/client.rb:27]: Hmm well, okay

File: src/api/app/services/trigger_controller_service/scm_extractor.rb (hunk 3 above)
Comment (krauselukas) [reply src/api/app/services/trigger_controller_service/scm_extractor.rb:2]: I will move it :)

File: src/api/app/apis/gitea_api/v1/client.rb (hunk 2 above)
Comment (krauselukas) [reply src/api/app/apis/gitea_api/v1/client.rb:1]: Yeah, maybe a bit to much to create a new folder in `app` for that, I will move it to the `lib` folder

**Review by danidoni — APPROVED** [review]

**Comment by danidoni** [comment]
We need to start thinking about refactoring the SCM vendors, otherwise we're not gonna get rid of the codeclimate issues...

**Comment by krauselukas** [comment]
Going to merge this, Hennes review was addressed

**Pull request merged**